flow/connectors/snowflake/snowflake.go

```diff
@@ -739,7 +739,11 @@ func (c *SnowflakeConnector) RenameTables(ctx context.Context, req *protos.Renam
 
 	if req.SyncedAtColName != nil {
 		for _, renameRequest := range req.RenameTableOptions {
-			resyncTblName := renameRequest.CurrentName
+			resyncSchemaTable, err := utils.ParseSchemaTable(renameRequest.CurrentName)
```

Comment:
invert the for loop and if statements so the conditionality happens inside the for loops, we can then reuse the parsed names for the entire loop instead of generating it 3 times per table

(yes this means we need to keep checking for the conditions, but I feel it would be more efficient than parsing the same thing thrice)